axelor-base/src/main/resources/domains/RegistrationNumberTemplate.xml

```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<domain-models xmlns="http://axelor.com/xml/ns/domain-models"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://axelor.com/xml/ns/domain-models http://axelor.com/xml/ns/domain-models/domain-models_6.1.xsd">
+
+  <module name="base" package="com.axelor.apps.base.db"/>
+
+  <entity name="RegistrationNumberTemplate">
+    <boolean name="isRequiredForCompanies" title="Required for companies" required="true"/>
+    <string name="titleToDisplay" namecolumn="true" required="true"/>
```

Comment:
it's not required if the user don't want to override the title, just keep the default title
And it's not the namecolumn, I forgot to ask you to add a name field in RegistrationNumberTemplate